design-system/materials/internals/story/shared-components.js

```diff
@@ -0,0 +1,270 @@
+                    onClick={(event) => {
+                      event.preventDefault();
+                      window.scrollTo({
+                        top: document.getElementById(
+                          `${props.id}-${configGroup.prefix}`
+                        ).offsetTop,
+                        behavior: 'smooth',
+                      });
+                    }}
+                  >
+                    {configGroup.label}
+                  </a>
+                </li>
+              )
+          )}
+        </ul>
+      )}
+    </>
+  );
+};
+TokenGroupLinks.propTypes = {
+  id: PropTypes.string.isRequired,
+  config: PropTypes.object,
+  filterText: PropTypes.string,
+  children: PropTypes.node.isRequired,
+};
+
+const TableBody = (props) => {
+  return (
+    <tbody>
+      {props.groupItems.map(([tokenName, tokenData]) => {
+        return (
+          <tr key={`${props.groupItemsPrefix}-${tokenName}-body-row-key`}>
+            {props.columnsConfig.map((columnConfig) => (
+              <td
+                key={`${props.groupItemsPrefix}-${tokenName}-${columnConfig.key}-body-cell-key`}
+              >
+                {props.cellRenderer({
+                  columnKey: columnConfig.key,
+                  groupItemsPrefix: props.groupItemsPrefix,
+                  tokenName,
+                  tokenData,
+                  themeName: props.themeName,
+                })}
+              </td>
+            ))}
+          </tr>
+        );
+      })}
+    </tbody>
+  );
+};
+TableBody.propTypes = {
+  groupItems: PropTypes.array.isRequired,
+  groupItemsPrefix: PropTypes.string,
+  themeName: PropTypes.string.isRequired,
+  columnsConfig: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      label: PropTypes.string,
+    })
+  ).isRequired,
+  cellRenderer: PropTypes.func.isRequired,
+};
+
+const TokensDetailsTable = (props) => (
+  <Table>
+    <thead>
+      <tr>
+        {props.columnsConfig.map((columnsConfig, index) => (
+          <td
+            key={columnsConfig.key}
+            style={{
+              minWidth: index === 0 ? 400 : 'auto',
+            }}
+          >
+            {columnsConfig.label || upperFirst(columnsConfig.key)}
+          </td>
+        ))}
+      </tr>
+    </thead>
+    <TableBody
+      groupItems={props.tokensGroupData}
+      groupItemsPrefix={props.tokensGroupPrefix}
+      themeName={props.themeName}
+      columnsConfig={props.columnsConfig}
+      cellRenderer={props.cellRenderer}
+    />
+  </Table>
+);
+TokensDetailsTable.propTypes = {
+  columnsConfig: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      label: PropTypes.string,
+    })
+  ).isRequired,
+  cellRenderer: PropTypes.func.isRequired,
+  themeName: PropTypes.string.isRequired,
+  tokensGroupPrefix: PropTypes.string,
+  tokensGroupData: PropTypes.array.isRequired,
+};
+
+export function SingleTokensGroupDetails(props) {
```

Comment:
This component is used for simple design token groups: states, component groups and variants.